alternator/streams.cc

```diff
@@ -673,9 +665,12 @@ future<executor::request_return_type> executor::get_shard_iterator(client_state&
         sid = rjson::get<shard_id>(request, "ShardId");
     } catch (...) {
     }
-    if (!schema || !cdc::get_base_table(db, *schema) || !is_alternator_keyspace(schema->ks_name()) || sid->table != schema->id()) {
+    if (!schema || !cdc::get_base_table(db, *schema) || !is_alternator_keyspace(schema->ks_name())) {
         throw api_error::resource_not_found("Invalid StreamArn");
     }
+    if (!sid) {
+        throw api_error::resource_not_found("Invalid ShardId");
+    }
```

Comment:
I wanted to suggest that you add tests which try to pass invalid shard ids or arns, and check we get ResourceNotFound as expected (and run the same test against DynamoDB to confirm that this is the right error to return in this case). But when I tried to look, I found a test you already wrote, test_get_shard_iterator(), which seems to suggest that a bad ARN should throw ValidationException, not ResourceNotFound... What am I missing?

Reply:
ARN == table id. `sid` above is shard id. That should generate resourcenotfound. And is tested.